packages/app/src/debug/StatsMetadata.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <ul
+    data-cy="stats-metadata"
+    class="flex flex-row items-center gap-x-2 text-gray-700 whitespace-nowrap children:flex children:items-center font-normal text-sm w-full"
+  >
+    <li
+      v-for="(result, i) in results"
+      :key="i"
+      :data-cy="`metaData-Results-${result.name}`"
+    >
+      <span
+        v-if="(result.value && (result.name === 'browser' || result.name === 'browser-groups'))"
+        class="flex inline-flex items-center"
+      >
+        <LayeredBrowserIcon
+          :order="result.icon"
+          :data-cy="`${result.name} ${result.value}`"
+        />
+        <span class="sr-only">{{ result.name }}</span>
+        {{ result.value }}
+      </span>
+      <span
+        v-else-if="result.value"
+        class="flex inline-flex items-center"
+      >
+        <component
+          :is="result.icon"
+          class="mr-8px"
+          stroke-color="gray-500"
+          fill-color="gray-100"
+          :data-cy="`${result.name} ${result.value}`"
+        />
+        <span class="sr-only">{{ result.name }}</span>
+        {{ result.value }}
+      </span>
+    </li>
+  </ul>
+</template>
+
+<script lang="ts" setup>
+
+import { computed } from 'vue'
+import type { SpecDataAggregate, CloudRunGroup } from '@packages/data-context/src/gen/graphcache-config.gen'
+import type { TestingTypeEnum } from '../generated/graphql'
+import { IconTimeStopwatch,
+  IconOsLinux,
+  IconOsApple,
+  IconOsGeneric,
+  IconOsWindows,
+  IconTestingTypeComponent,
+  IconTestingTypeE2E,
+  IconTechnologyServer,
+} from '@cypress-design/vue-icon'
+
+import LayeredBrowserIcon from './LayeredBrowserIcons.vue'
+
+type StatType = 'DURATION' | 'OS' | 'BROWSER' | 'TESTING' | 'G_OS' | 'GROUPS' | 'G_BROWSERS' | 'GROUP_NAME'
+
+interface MetadataProps {
+  order?: StatType[]
+  specDuration?: string | number
+  testing?: TestingTypeEnum
+  groups?: CloudRunGroup[]
+  groupName?: string
+}
+
+const props = defineProps<MetadataProps>()
+
+interface Metadata {
+  value: number | string | null | SpecDataAggregate | TestingTypeEnum
+  icon: any
+  name: string
+}
+
+type OSType = 'LINUX' | 'APPLE' | 'WINDOWS' | 'GROUP'
+
+const OS_MAP: Record<OSType, any> = {
+  'LINUX': IconOsLinux,
+  'APPLE': IconOsApple,
+  'WINDOWS': IconOsWindows,
+  'GROUP': IconOsGeneric,
+}
+
+const TESTING_MAP: Record<TestingTypeEnum, any> = {
+  'e2e': IconTestingTypeE2E,
+  'component': IconTestingTypeComponent,
+}
+
+const results = computed(() => {
+  if (props.order) {
+    return props.order.map((status) => ORDER_MAP[status])
+  }
+
+  return []
+})
+
+const arrMapping = computed(() => {
+  const acc: {browser: string[], os: string[], firstBrowser: string, firstOS: string} = { browser: [], os: [], firstBrowser: '', firstOS: '' }
+  const uniqueBrowser: Set<string> = new Set()
+  const uniqueOS: Set<string> = new Set()
```

Comment:
If you want an array of unique values you can do it like this

```js
m = new Map()
m.set('foo', { name: 'foo' })
m.set('foo', { name: 'foo' })
m.set('bar', { name: 'bar' })

const arr = Array.from(m.values())
```

This was, you do not need a `Set` and an Array.

Reply:
(Or am I missing something? I may have misunderstood what's going on here).

Reply:
I don't think you are missing anything. This method was made to obtain unique Os and Browser values. Thank you for the `Array.from(m.values())`